internal/wsutil/wsutil_linux.go

```diff
@@ -26,42 +26,31 @@ import (
 	"github.com/canonical/pebble/internal/logger"
 )
 
-// WebsocketExecMirror mirrors a websocket connection with a set of Writer/Reader.
-func WebsocketExecMirror(conn MessageReadWriter, w io.WriteCloser, r io.ReadCloser, exited chan struct{}, fd int) (chan bool, chan bool) {
-	readDone := make(chan bool, 1)
-	writeDone := make(chan bool, 1)
-
-	go DefaultWriter(conn, w, writeDone)
-
-	go func(conn MessageWriter, r io.ReadCloser) {
-		in := ExecReaderToChannel(r, -1, exited, fd)
-		for {
-			buf, ok := <-in
-			if !ok {
-				r.Close()
-				logger.Debugf("Sending write barrier")
-				err := conn.WriteMessage(websocket.TextMessage, endCommandJSON)
-				if err != nil {
-					logger.Debugf("Got err writing barrier %s", err)
-				}
-				readDone <- true
-				return
-			}
-
-			err := conn.WriteMessage(websocket.BinaryMessage, buf)
+// MirrorToWebsocket mirrors PTY output from r (file descriptor fd) to the websocket.
+func MirrorToWebsocket(conn MessageWriter, r io.ReadCloser, exited chan struct{}, fd int) {
+	in := ExecReaderToChannel(r, -1, exited, fd)
+	for {
+		buf, ok := <-in
+		if !ok {
+			r.Close()
+			logger.Debugf("Sending write barrier")
+			err := conn.WriteMessage(websocket.TextMessage, endCommandJSON)
 			if err != nil {
-				logger.Debugf("Got err writing %s", err)
-				break
+				logger.Debugf("Got err writing barrier %s", err)
```

Comment:
This was already an issue before in this file, and we are probably inheriting this code from elsewhere, but for the record these log messages aren't great as they have no context to them, and even knowing the context they feel somewhat unclear (e.g. "Got err writing" below.. writing what, to what, when, etc).

Just a note. It was already like this before and not worth going around fixing everything in this PR, but would be nice to clean it up eventually.